inference-engine/src/preprocessing/cpu_x86_avx2/ie_preprocess_gapi_kernels_avx2.cpp

```diff
@@ -131,6 +131,387 @@ void calcRowArea_32F(float dst[], const float *src[], const Size& inSz,
+                    val_2 = v_insert64< 2 >(val_2, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 10]]));
+                    val_2 = v_insert64< 3 >(val_2, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 11]]));
+
+                    val_3 = v_insert64< 0 >(val_3, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 12]]));
+                    val_3 = v_insert64< 1 >(val_3, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 13]]));
+                    val_3 = v_insert64< 2 >(val_3, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 14]]));
+                    val_3 = v_insert64< 3 >(val_3, *reinterpret_cast<int64_t*>(&tmp[4 * mapsx[x + 15]]));
+
+                    val_0 = v_permutevar8x32(val_0, idxs);
+                    val_1 = v_permutevar8x32(val_1, idxs);
+                    val_2 = v_permutevar8x32(val_2, idxs);
+                    val_3 = v_permutevar8x32(val_3, idxs);
+
+                    v_int16 val0_0 = v_reinterpret_as_s16(v_expand_low(val_0));
+                    v_int16 val0_1 = v_reinterpret_as_s16(v_expand_low(val_1));
+                    v_int16 val0_2 = v_reinterpret_as_s16(v_expand_low(val_2));
+                    v_int16 val0_3 = v_reinterpret_as_s16(v_expand_low(val_3));
+
+                    v_int16 val1_0 = v_reinterpret_as_s16(v_expand_high(val_0));
+                    v_int16 val1_1 = v_reinterpret_as_s16(v_expand_high(val_1));
+                    v_int16 val1_2 = v_reinterpret_as_s16(v_expand_high(val_2));
+                    v_int16 val1_3 = v_reinterpret_as_s16(v_expand_high(val_3));
+
+                    v_int16 t0 = v_mulhrs(v_sub_wrap(val0_0, val1_0), a10);
+                    v_int16 t1 = v_mulhrs(v_sub_wrap(val0_1, val1_1), a32);
+                    v_int16 t2 = v_mulhrs(v_sub_wrap(val0_2, val1_2), a54);
+                    v_int16 t3 = v_mulhrs(v_sub_wrap(val0_3, val1_3), a76);
+
+                    v_int16 r0 = v_add_wrap(val1_0, t0);
+                    v_int16 r1 = v_add_wrap(val1_1, t1);
+                    v_int16 r2 = v_add_wrap(val1_2, t2);
+                    v_int16 r3 = v_add_wrap(val1_3, t3);
+
+                    v_uint8 q0 = v_packus(r0, r1);
+                    v_uint8 q1 = v_packus(r2, r3);
+
+                    v_uint8 q2 = v_shuffle_s8(q0, shuf_mask);
+                    v_uint8 q3 = v_shuffle_s8(q1, shuf_mask);
+
+                    v_uint8 q4 = v_blend_shiftleft<0xCC /*0b11001100*/, 4>(q2, q3);
+                    v_uint8 q5 = v_blend_shiftright<0xCC /*0b11001100*/, 4>(q2, q3);
+
+                    v_uint8 q6 = v256_permute4x64<0xD8>(q4);
+                    v_uint8 q7 = v256_permute4x64<0xD8>(q5);
+
+                    v_uint8 q8 = v_shuffle_s8(q6, shuf_mask1);
+                    v_uint8 q9 = v_shuffle_s8(q7, shuf_mask1);
+
+                    v_store_low(&dst[0][x], q8);
+                    v_store_high(&dst[1][x], q8);
+                    v_store_low(&dst[2][x], q9);
+                    v_store_high(&dst[3][x], q9);
+                }
```

Comment:
As far as I can see there is no much difference between this horizontal pass and the pass from `(!xRatioEq1 && !yRatioEq1)` branch so I propose to reuse this pass, not to duplicate it

Reply:
I cancelled out duplicate code to a function.